qt/applications/workbench/workbench/plotting/figurewindow.py

```diff
@@ -30,19 +30,26 @@ def _validate_workspaces(names: List[str]) -> List[bool]:
     :param names: A list of workspace names
     :return: List of bools, True if all workspaces have multiple bins
     """
+    ws = None
     ads = AnalysisDataServiceImpl.Instance()
     has_multiple_bins = []
     for name in names:
         result = None
-        ws = ads.retrieve(name)
+
+        try:
```

Comment:
I think `ws = None` should be within the loop. If the loop assigns ws to a workspace, and then the try block gets an error (so ws is not reset) the previous workspace will be used, hiding the problem.

Reply:
Got it, I have corrected it and push a fix for correction.